modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -40,10 +36,27 @@ type Processor struct {
 }
 
 func New(cfg Config, registry registry.Registry) gen.Processor {
-	labels := make([]string, 0, len(intrinsicDimensions)+len(cfg.Dimensions))
-	labels = append(labels, intrinsicDimensions...)
+	labels := make([]string, 0, 4+len(cfg.Dimensions))
+
+	if cfg.IntrinsicDimensions.Service {
+		labels = append(labels, dimService)
+	}
+	if cfg.IntrinsicDimensions.SpanName {
+		labels = append(labels, dimSpanName)
+	}
+	if cfg.IntrinsicDimensions.SpanKind {
+		labels = append(labels, dimSpanKind)
+	}
+	if cfg.IntrinsicDimensions.StatusCode {
+		labels = append(labels, dimStatusCode)
+	}
+	if cfg.IntrinsicDimensions.StatusMessage {
+		labels = append(labels, dimStatusMessage)
+	}
+	intrinsicDimensions := labels[:]
```

Comment:
why copy the slice here? it's pointing to the same underlying array.

Reply:
This was needed in order to detect collisions with configured intrinsic dimensions. This is no longer needed as dimensions are now also treated as collisions when the colliding intrinsic dimension is absent.